configs/components/rubygem-ffi.rb

```diff
@@ -1,18 +1,33 @@
 component "rubygem-ffi" do |pkg, settings, platform|
   instance_eval File.read('configs/components/base-rubygem.rb')
-  pkg.version "1.9.18"
 
-  if platform.is_windows?
-    if platform.architecture == "x64"
-      pkg.md5sum "664afc6a316dd648f497fbda3be87137"
-      pkg.url "https://rubygems.org/downloads/ffi-#{pkg.get_version}-x64-mingw32.gem"
-      pkg.mirror "#{settings[:buildsources_url]}/ffi-#{pkg.get_version}-x64-mingw32.gem"
+  version = settings[:rubygem_ffi_version] || '1.9.18'
+  pkg.version version
+
+  case version
+    when '1.9.14'
+      if platform.architecture == "x64"
+        pkg.md5sum "6d78ae5b2378513d3fc68600981366e9"
+        pkg.url "https://rubygems.org/downloads/ffi-#{pkg.get_version}-x64-mingw32.gem"
+      else
+        pkg.md5sum "54a20f0f73d03c06adf63789faa53746"
+        pkg.url "https://rubygems.org/downloads/ffi-#{pkg.get_version}-x86-mingw32.gem"
+      end
+    when '1.9.18'
+      if platform.architecture == "x64"
+        pkg.md5sum "664afc6a316dd648f497fbda3be87137"
+        pkg.url "https://rubygems.org/downloads/ffi-#{pkg.get_version}-x64-mingw32.gem"
+        pkg.mirror "#{settings[:buildsources_url]}/ffi-#{pkg.get_version}-x64-mingw32.gem"
+      else
+        pkg.md5sum "0b6fd994826952231d285f078cefce32"
+        pkg.url "https://rubygems.org/downloads/ffi-#{pkg.get_version}-x86-mingw32.gem"
+        pkg.mirror "#{settings[:buildsources_url]}/ffi-#{pkg.get_version}-x86-mingw32.gem"
```

Comment:
The url and mirror stuff could get moved out of the case statement, since that uses pkg.get_version anyways, just needs to switch on architecture